lib/streaming-pipeline.js

```diff
@@ -131,18 +165,31 @@ module.exports = class StreamingPipeline extends Writable {
                 callback(new RiffError(
                     'error-argument-transformer',
                     `Function must declare exactly ${inputCount} argument transformer(s). Found ${transformerCount}`));
-                return false;
+                return;
             }
+
+            const negotiatedOutputContentTypes = outputContentTypes.map(ct => determineContentTypes(ct).accept);
+            for (let i = 0; i < negotiatedOutputContentTypes.length; i++) {
```

Comment:
it's odd to do a map on an array and then immediately iterate over the resulting array. Any reason these can't be combined?

You might also be interested in a [for...of loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)

Reply:
done with `findIndex`

Reply:
it could even be better with an Option type and a `fold` function 🤷🏻‍♂️

Reply:
hmm, thinking about it, the only way to iterate only is to use a for loop (I cannot use for..of because I need the index)

Reply:
now fixed